packages/relay-compiler/bin/RelayCompilerMain.js

```diff
@@ -174,86 +176,107 @@ Root files can include:
 - A .hg/ Mercurial folder
 - A .watchmanconfig file
 
-Ensure that one such file exists in ${srcDir} or its parents.
+Ensure that one such file exists in ${src} or its parents.
     `.trim(),
     );
   }
   if (options.verbose && options.quiet) {
     throw new Error("I can't be quiet and verbose at the same time");
   }
 
+  const watchman = options.watchman && (await WatchmanClient.isAvailable());
+
+  const codegenRunner = getCodegenRunner({
+    ...options,
+    persistOutput,
+    schema,
+    src,
+    watchman,
+  });
+
+  if (!options.validate && !options.watch && watchman) {
+    // eslint-disable-next-line no-console
+    console.log('HINT: pass --watch to keep watching for changes.');
+  }
+
+  const result = options.watch
+    ? await codegenRunner.watchAll()
+    : await codegenRunner.compileAll();
+
+  if (result === 'ERROR') {
+    process.exit(100);
+  }
+  if (options.validate && result !== 'NO_CHANGES') {
+    process.exit(101);
+  }
+}
```

Comment:
Most of these changes are to refactor configuration of the codegen runner out into a standalone function that can be tested.